tests/integration_tests/functional/test_cpu_features_host_vs_guest.py

```diff
@@ -0,0 +1,270 @@
+# Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# pylint: disable=too-many-statements
+
+"""
+Check CPU features in the host vs the guest.
+
+This test can highlight differences between the host and what the guest sees.
+
+No CPU templates as we are interested only on what is passed through to the guest by default.
+For that, check test_feat_parity.py
+"""
+
+import os
+
+from framework import utils
+from framework.properties import global_props
+from framework.utils_cpuid import CPU_FEATURES_CMD, CpuModel
+
+CPU_MODEL = global_props.cpu_codename
+
+INTEL_HOST_ONLY_FEATS = {
+    "acpi",
+    "aperfmperf",
+    "arch_perfmon",
+    "art",
+    "bts",
+    "cat_l3",
+    "cdp_l3",
+    "cqm",
+    "cqm_llc",
+    "cqm_mbm_local",
+    "cqm_mbm_total",
+    "cqm_occup_llc",
+    "dca",
+    "ds_cpl",
+    "dtes64",
+    "dtherm",
+    "dts",
+    "epb",
+    "ept",
+    "ept_ad",
+    "est",
+    "flexpriority",
+    "flush_l1d",
+    "hwp",
+    "hwp_act_window",
+    "hwp_epp",
+    "hwp_pkg_req",
+    "ida",
+    "intel_ppin",
+    "intel_pt",
+    "mba",
+    "monitor",
+    "pbe",
+    "pdcm",
+    "pebs",
+    "pln",
+    "pts",
+    "rdt_a",
+    "sdbg",
+    "smx",
+    "tm",
+    "tm2",
+    "tpr_shadow",
+    "vmx",
+    "vnmi",
+    "vpid",
+    "xtpr",
+}
+
+
+def test_host_vs_guest_cpu_features(uvm_nano):
+    """Check CPU features host vs guest"""
+
+    vm = uvm_nano
+    vm.add_net_iface()
+    vm.start()
+    host_feats = set(utils.check_output(CPU_FEATURES_CMD).stdout.split())
+    guest_feats = set(vm.ssh.check_output(CPU_FEATURES_CMD).stdout.split())
+
+    match CPU_MODEL:
+        case CpuModel.AMD_MILAN:
+            host_guest_diff_5_10 = {
```

Comment:
nit: The variable names of the expected feat sets are different between cases. How about unifying the variable name (like `host_minus_guest_5_10`, `host_only_5_10`, or something)?